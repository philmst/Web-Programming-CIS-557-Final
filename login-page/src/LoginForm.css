form {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    height: 300px;
}

.textDiv {
    display: flex;
    flex-direction: column;
}

input[type="submit"] {
    align-self: center;
}

.text {
    width: 200px;
    height: 25px;
}

.LoginForm {
    border: 0.5px solid rgba(128,128,128, 0.2);
    border-radius: 5px;
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 500px;
    width: 500px;
    justify-content: space-around;
}