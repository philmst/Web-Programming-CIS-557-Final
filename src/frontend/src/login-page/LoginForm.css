form {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    height: 300px;
}

.textDiv {
    display: flex;
    flex-direction: column;
}

input[type="submit"] {
    align-self: center;
    color: #FEFCFB;
    background-color: #1282A2;
}

.text {
    width: 200px;
    height: 25px;
}

.LoginForm, .RegistrationForm {
    border: 0.5px solid rgb(0,31,84);
    border-radius: 5px;
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 500px;
    width: 500px;
    justify-content: space-around;
}

h1 {
    color: #034078;
}