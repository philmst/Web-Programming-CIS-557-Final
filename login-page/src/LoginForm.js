import './LoginForm.css';

function LoginForm() {
  return (
    <div className="LoginForm">
      <h1>Welcome!</h1>
      <form>
        <div className="textDiv" id="usernameDiv">
          <label for="username">Username:</label>
          <input className="text" id="username" type="text" placeholder="username"/>
        </div>
        <div className="textDiv" id="passwordDiv">
          <label for="password">Password:</label>
          <input className="text" id="password" type="password" placeholder="password"/>
        </div>
        <input type="submit" value="Login"/>
      </form>
    </div>
  );
}

export default LoginForm;
