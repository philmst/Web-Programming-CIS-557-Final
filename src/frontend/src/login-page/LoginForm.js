import React from 'react';
import { Link } from 'react-router-dom';
import './LoginForm.css';

const lib = require('./LoginModule');

function LoginForm(props) {
  const { changeLink } = props;

  return (
    <div className="LoginForm">
      <h1>Welcome!</h1>
      {window.location.href.split('/').pop() === 'error' ? (<p>Username or password does not exist.</p>) : null}
      <div className="textDiv" id="usernameDiv">
        <label htmlFor="username">
          Username:
          <input className="text" id="username" type="text" placeholder="username" />
        </label>
      </div>
      <div className="textDiv" id="passwordDiv">
        <label htmlFor="password">
          Password:
          <input className="text" id="password" type="password" placeholder="password" />
        </label>
      </div>
      <p>
        Not a user?
        <Link id="registrationLink" to="/registration" onClick={() => changeLink('/registration')}> Create an account. </Link>
      </p>
      <Link to="/main" onClick={() => lib.verifyUser(changeLink, document.getElementById('username').value, document.getElementById('password').value)}>
        <input id="loginButton" type="submit" value="Login" />
      </Link>
    </div>
  );
}

export default LoginForm;
