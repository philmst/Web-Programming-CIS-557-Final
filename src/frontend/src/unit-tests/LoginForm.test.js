import React from 'react';
import { render } from '@testing-library/react';
import LoginForm from '../login-page/LoginForm';
import { BrowserRouter as Router } from 'react-router-dom';

/**
 * @jest-environment jsdom
 */

const changeLink = jest.fn();

describe('fetch and DOM testing with mocking', () => {
  test('change link called when login clicked', async () => {
    render(<Router><LoginForm changeLink={changeLink} /></Router>);
    document.getElementById("loginButton").click();
    expect(changeLink).toBeCalled();
    expect(changeLink.mock.calls[0][0]).toBe('/main');
  });

  test('change link called when registration link clicked', async () => {
    render(<Router><LoginForm changeLink={changeLink} /></ Router>);
    document.getElementById("registrationLink").click();
    expect(changeLink).toBeCalled();
    expect(changeLink.mock.calls[0][0]).toBe('/registration');
  });
});
