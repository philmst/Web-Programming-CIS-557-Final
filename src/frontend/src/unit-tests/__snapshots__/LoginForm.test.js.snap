// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch and DOM testing with mocking Login page renders correctly 1`] = `
<div
  className="LoginForm"
>
  <h1>
    Welcome!
  </h1>
  <div
    className="textDiv"
    id="usernameDiv"
  >
    <label
      htmlFor="username"
    >
      Username:
      <input
        className="text"
        id="username"
        placeholder="username"
        type="text"
      />
    </label>
  </div>
  <div
    className="textDiv"
    id="passwordDiv"
  >
    <label
      htmlFor="password"
    >
      Password:
      <input
        className="text"
        id="password"
        placeholder="password"
        type="password"
      />
    </label>
  </div>
  <p>
    Not a user?
    <a
      href="/registration"
      id="registrationLink"
      onClick={[Function]}
    >
       Create an account. 
    </a>
  </p>
  <a
    href="/main"
    onClick={[Function]}
  >
    <input
      id="loginButton"
      type="submit"
      value="Login"
    />
  </a>
</div>
`;
